libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/summary/InstructionListPresenter.java

```diff
@@ -157,7 +157,7 @@ private boolean removeInstructionsFrom(int currentInstructionIndex) {
       instructions.remove(FIRST_INSTRUCTION_INDEX);
       return true;
     } else if (currentInstructionIndex <= instructions.size()) {
-      instructions.subList(FIRST_INSTRUCTION_INDEX, currentInstructionIndex).clear();
+      instructions.subList(FIRST_INSTRUCTION_INDEX, currentInstructionIndex + 1).clear();
```

Comment:
I don't think this is the root cause. I've tested legacy test app (`base-v0.42.1` branch) which is implementing `instructions.subList(FIRST_INSTRUCTION_INDEX, currentInstructionIndex).clear();` and it's not showing that duplicated stale instruction. So pretty sure is something else that introduced the regression. It seems that something changed around the indices logic.